pandas/core/internals/blocks.py

```diff
@@ -2114,10 +2114,59 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
+    @cache_readonly
+    def is_numeric(self) -> bool:
+        dtype = self.values.dtype
+        kind = dtype.kind
 
-class NumericBlock(NumpyBlock):
-    __slots__ = ()
-    is_numeric = True
+        if kind in "fciub":
+            return True
+        else:
+            return False
+
+    @cache_readonly
+    def is_object(self) -> bool:
+        if self.values.dtype.kind == "O":
+            return True
+        else:
+            return False
+
+    @maybe_split
```

Comment:
im pretty sure we don't want the maybe_split in non-object cases, so will need to do this without the decorator